Add Fringe Weaver, Sugar Rush, Crevice Spike drinks from Va-11 Hall-A

BunnyBot5000 opened the pull request:
[Changelogs]: # (Your PR should contain a detailed changelog of notable changes, titled and categorized appropriately. This includes, new features, sprites, sounds, balance changes, admin tools, map edits, removals, big refactors, config changes, hosting changes and important fixes. An example changelog has been provided below for you to edit. If you need additional help, read https://github.com/tgstation/tgstation/wiki/Changelogs)

:cl:
add: Fringe Weaver drink - made with ethanol and sugar, it's classy (and slightly weaker) hooch
add: Sugar Rush drink - made with sugar, lemon juice, and wine. It's got a slight nutritional value but decreases your satiety, causing you to be unable to consume junk food.
add: Crevice Spike drink - made with lime juice and hot sauce. A 'sobering' drink, the Crevice Spike will reduce your drunkenness slightly but it kicks hard going down, dealing 15 points of brute when first ingested.
/:cl:

I was a bit disappointed to find that the space bartender doesn't have any options from Va-11 Hall-A's drink menu to serve, so I cooked up my own versions of a few of the drinks. 

In Va-11 Hall-A, the Fringe Weaver has the highest alcohol content of any drink, described as 'ethylic alcohol with a spoonful of sugar', so I wanted it to be reflected as a high power drink.

The Sugar Rush is the easiest drink to brew in Valhalla, so I wanted its ingredients to be easy to acquire (it's all in the booze-o-mat / the softdrink machine), and it's practically junk food.

The Crevice Spike is the most 'out there' of the three drinks - it has a negative booze power so it'll slowly sober you up, but it kicks hard going down.

- **Fringe Weaver:** 1 part sugar, 9 parts ethanol
![fringe weaver dark glass](https://user-images.githubusercontent.com/37563222/37631882-cc4570a6-2ba8-11e8-9263-213093ea9afd.png)
- **Sugar Rush:** 2 parts sugar, 1 part lemon juice, 1 part wine
![sugar rush](https://user-images.githubusercontent.com/37563222/37631919-f9c28e9c-2ba8-11e8-842d-cbafdc2436b0.png)
- **Crevice Spike:** 2 parts lime juice, 4 parts hot sauce / capsaicin
![crevice spike](https://user-images.githubusercontent.com/37563222/37631934-0d372ff0-2ba9-11e8-847c-05f21fe7d583.png)

BunnyBot5000 committed:
Add Fringe Weaver, Sugar Rush, Crevice Spike from Va-11 Hall-A

francinum commented on code/modules/reagents/chemistry/reagents/alcohol_reagents.dm:
```diff
@@ -1505,3 +1505,46 @@ All effects don't start immediately, but rather get worse over time; the rate is
+	glass_desc = "It'll either knock the drunkenness out of you or knock you out cold. Both, probably."
+
+/datum/reagent/consumable/ethanol/crevice_spike/on_mob_add(mob/living/L)
+	L.adjustBruteLoss(15) //That first gulp hurts going down! But you can finish the drink much easier on your gut-zone.
```
This is going to be abused to an insane degree, do we have any other chems that do straight damage like this?

BunnyBot5000 replied:
Would it be better to reduce the damage on first ingestion (it only deals damage the first time it enters your system, and won't again until its out), or find some other means of stunning / harming the drinker to offset the sobering effect?

MrDoomBringer replied:
remember it's only on the first /sip/ of it. Not certain how it could be abused to an insane degree, as its only useful once per person (until it metabolizes out i guess)

BunnyBot5000 replied:
@MrDoomBringer when I was testing this (and figuring out the code) I accidentally made it do 2 brute damage per tick while it metabolized - suffice to say, it killed the drinker really fast.

fludd12 replied:
I think the primary concern is that people might inject 0.1 units at a time, once every two seconds, to deal damage a lot faster than any conventional chem. Considering the finnicky-ness of doing something like that, though, I doubt it'll be too concerning.

BunnyBot5000 replied:
@fludd12 I feel that if you've got someone in a position where you can inject them repeatedly with crevice spike over time without them noticing / panicking, you've got a person in a position where you can do just about anything to them.

fludd12 replied:
My exact same thought on the subject!

francinum replied:
I meant it more as a cheap additive for syringe gun mixes, but I think the new change fixes it.

MrDoomBringer commented:
look like pretty nice sprites!

BunnyBot5000 commented:
Thanks! I tried to make them as true to the Va-11 Hall-A sprites as I could.

BunnyBot5000 committed:
Rebalance crevice spike damage

BunnyBot5000 commented:
Alright, I mucked with the Crevice Spike code - it will now deal 3 damage per unit ingested up to a max 5 units / 15 damage. A single 5 unit gulp will truck you for 15, but smaller amounts will hurt less.

Tlaltecuhtli commented:
sugar rush has no sugar?

ExcessiveUseOfCobblestone commented on code/modules/reagents/chemistry/reagents/alcohol_reagents.dm:
```diff
@@ -1505,3 +1505,47 @@ All effects don't start immediately, but rather get worse over time; the rate is
+
+/datum/reagent/consumable/ethanol/crevice_spike/on_mob_add(mob/living/L) //damage only applies when drink first enters system and won't again until drink metabolizes out
+	var/initialburn = 5
+	for(var/i in 1 to min(volume, initialburn)) //minimum 3 brute damage on ingestion to limit non-drink means of injury - a full 5 unit gulp of the drink trucks you for the full 15
```
what's the point of the var if there's no way to edit it?

Cyberboss commented:
>no bad touch

why even contribute?

Cyberboss commented on code/modules/reagents/chemistry/reagents/alcohol_reagents.dm:
```diff
@@ -1505,3 +1505,47 @@ All effects don't start immediately, but rather get worse over time; the rate is
+/datum/reagent/consumable/ethanol/crevice_spike/on_mob_add(mob/living/L) //damage only applies when drink first enters system and won't again until drink metabolizes out
+	var/initialburn = 5
+	for(var/i in 1 to min(volume, initialburn)) //minimum 3 brute damage on ingestion to limit non-drink means of injury - a full 5 unit gulp of the drink trucks you for the full 15
+		L.adjustBruteLoss(3)
```
condense these 3 lines to L.adjustBruteLoss(3 * min(5, volume))

Cyberboss commented on code/modules/food_and_drinks/recipes/drinks_recipes.dm:
```diff
@@ -603,3 +603,24 @@
+	name = "Sugar Rush"
+	id = "sugar_rush"
+	results = list("sugar_rush" = 4)
+	required_reagents = list("sodawater" = 2, "lemonjuice" = 1, "wine" = 1) //2 adelhyde (sweet), 1 powdered delta (sour), 1 karmotrine (alcohol)
```
sugar rush needs sugar yo

BunnyBot5000 committed:
Crevice Spike cleanup and sugar-y Sugar Rush

BunnyBot5000 commented:
Cleaned up the Crevice Spike code and replaced the Sugar Rush's soda water with sugar.

> no bad touch
If this gets merged I'll probably do a second round after the feature freeze is over, heh heh. The positive reception of this pr has been pretty inspiring.

ExcessiveUseOfCobblestone commented:
meow

PKPenguin321 commented:
will vaping crevice spike melt peoples faces off

ExcessiveUseOfCobblestone commented:
only if you do it in such low amounts that it constantly gets removed and added, and if you're doing that the volume would be so low they'd basically have to be not moving to take substantial damage.

optimumtact commented:
welcome aboard
